### PR TITLE
MMA-17737: add cp-all-in-one verification block to PR pipeline

### PR DESCRIPTION
**Pull request by riagarwal2002**
## Summary

Automates the manual screenshot step from [.github/PULL_REQUEST_TEMPLATE.md](.github/PULL_REQUEST_TEMPLATE.md) so a broken docker image (cf. INC-6655 for C3 2.3.0) can't merge silently. Closes [MMA-17737](https://confluentinc.atlassian.net/browse/MMA-17737).

### What the new block does

`.semaphore/semaphore.yml` gets a new "Verify cp-all-in-one Docker Compose" block that runs on every PR after `Build, Test, & Scan AMD`. The actual logic lives in [.semaphore/scripts/verify-cp-all-in-one.sh](.semaphore/scripts/verify-cp-all-in-one.sh) (so it's easy to read, lint, and run locally outside CI), and the YAML block just invokes it with `up` and `down` subcommands. The script:

1. **Derives the matching cp-all-in-one branch** from this repo's `pom.xml` parent `<version>` (e.g. `[8.0.4-0, 8.0.5-0)` → `8.0.4`). The pom is the exact CP version the images are built against, so it's the authoritative source. Resolution order on cp-all-in-one:
   - Prefer `<cp-version>-post` (released-patch pin).
   - Else `<cp major.minor>.x` (active-dev branch — used when `-post` hasn't been cut yet but the CP line is in development).
   - Else `master` as last resort.
2. **Clones** the resolved branch of `confluentinc/cp-all-in-one`.
3. **Sed-patches** the three image refs (`cp-enterprise-control-center-next-gen`, `cp-enterprise-prometheus`, `cp-enterprise-alertmanager`) in `docker-compose.yml` to point at the PR's dev ECR tags. All other images (cp-server, schema-registry, ksqldb, etc.) are left untouched and pull from Docker Hub at their pinned CP version.
4. **`docker compose pull` then `docker compose up -d`** the whole stack; polls `:9021/`, `:9090/-/ready`, `:9093/-/ready` for HTTP 200 — up to 10 minutes (60 × 10s) — and fails the PR if any service doesn't come up.
5. **Tears down** with `docker compose ps` + per-service logs + `down -v` in the always-epilogue.

This would have caught INC-6655 (broken template file → C3 fails to start → `:9021/` never returns 200).

### Stale comment cleanup

Also removes the "managed by ServiceBot" header comment from all three `.semaphore/*.yml` files. This repo opted out of ServiceBot pipeline management in [4249d1b](https://github.com/confluentinc/control-center-next-gen-images/commit/4249d1b66a07b876ec2b6529a6666712f648958b) (Apr 2025) via `pipeline_enable: false` in `service.yml`, so the comment has been misleading for over a year. @krishvora01 (CPBR) confirmed in Slack the comment can go.

## Test plan

- [x] CI on this PR passes the new block end-to-end. Job log confirms:
  - `Using cp-all-in-one branch: 8.0.4-post (CP_VERSION=8.0.4, from pom.xml parent.version)` — pom-driven derivation hit, `*-post` branch selected.
  - `grep` after sed shows exactly the three enterprise images point at PR-built ECR URLs (`dev-2.3.x-<sha>-ubi9.amd64`); the other 10 services keep their `:8.0.4` public tags.
  - Healthchecks reach 200 in ~50s — well under the 10-minute cap.
  - Total verification block time: ~6 min.
  
## Rollout

Landing on 2.3.x first (lowest active branch). Will be forward-merged via pint-merge to 2.4.x → 2.5.x → 2.6.x → master. The script reads `pom.xml` at runtime, so the same code self-configures correctly on each branch — no per-branch fixups expected.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[MMA-17737]: https://confluentinc.atlassian.net/browse/MMA-17737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .semaphore/cp_dockerfile_build.yml [hunk 1]
```diff
@@ -1,8 +1,3 @@
-# This file is managed by ServiceBot plugin - Semaphore. The content in this file is created using a common
```
Comment (rasikaAjoshi23) [inline .semaphore/cp_dockerfile_build.yml:1]: why are we removing this comment?

**Review by rasikaAjoshi23 — COMMENTED** [review]
One minor comment, we can merge and test the change overall LGTM

File: .semaphore/cp_dockerfile_build.yml (hunk 1 above)
Comment (riagarwal2002) [reply .semaphore/cp_dockerfile_build.yml:1]: This is because this repo opted out of ServiceBot pipeline management in [4249d1b](https://github.com/confluentinc/control-center-next-gen-images/commit/4249d1b66a07b876ec2b6529a6666712f648958b) (Apr 2025) via pipeline_enable: false in service.yml, so the comment has been misleading for over a year. It is not managed by service bot anymore.

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -159,6 +154,101 @@ blocks:
...
+            #   1. Exact C3_VERSION lookup (e.g. "2.3.2") -> CP_VERSION (e.g. "8.0.4")
+            #      Falls back to .x lookup (e.g. "2.3.x") if exact isn't in json.
+            #   2. Prefer "<CP_VERSION>-post" (the released-patch pin) on cp-all-in-one.
+            #   3. Else "<CP major.minor>.x" (active-dev branch for that CP line,
+            #      used when -post hasn't been cut yet but the line is in dev).
+            #   4. Else "master" (last-resort fallback).
+            - |
+              C3_VERSIONS_JSON=$(gh api repos/confluentinc/depot/contents/python/common_tools/data/prod/c3_versions.json \
+                | jq -r '.content' | base64 -d)
+              CP_VERSION=$(echo "$C3_VERSIONS_JSON" \
```
Comment (KrishVora01) [inline .semaphore/semaphore.yml]: Would it be better to get this value from the pom.xml, since that is the exact version that will be used to build the images?
While the c3_versions.json way may work, however, if someone changes the value in the json file, but does not run the task to update the values in this repo, we might end up testing against a stale CP version entry.

File: .semaphore/semaphore.yml [hunk 3]
```diff
@@ -159,6 +154,101 @@ blocks:
...
+                C3_KEY="$(echo "$C3_VERSION" | grep -oE '^[0-9]+\.[0-9]+').x"
+                echo "No c3_versions.json entry for exact $C3_VERSION; trying $C3_KEY"
+                CP_VERSION=$(echo "$C3_VERSIONS_JSON" \
+                  | jq -r --arg v "$C3_KEY" '.[$v]["confluent.version"] // empty' \
+                  | sed -E 's/-cp[0-9]+$//')
+              fi
+              if [ -n "$CP_VERSION" ]; then
+                CP_POST_BRANCH="${CP_VERSION}-post"
+                CP_X_BRANCH="$(echo "$CP_VERSION" | grep -oE '^[0-9]+\.[0-9]+').x"
+                if git ls-remote --exit-code --heads https://github.com/confluentinc/cp-all-in-one.git "$CP_POST_BRANCH" >/dev/null 2>&1; then
```
Comment (KrishVora01) [inline .semaphore/semaphore.yml]: I'd suggest moving these commands to a new script in itself, imo it would be easier to debug if it were a standalone script, rather than all commands be present in the semaphore.yml itself.

Comment (riagarwal2002) [reply .semaphore/semaphore.yml]: sure, done

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (riagarwal2002) [reply .semaphore/semaphore.yml]: fair yes, changing to pom.xml

File: .semaphore/scripts/verify-cp-all-in-one.sh [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  local cp_post="${cp_version}-post"
+  local cp_x
+  cp_x="$(echo "$cp_version" | grep -oE '^[0-9]+\.[0-9]+').x"
+
+  if git ls-remote --exit-code --heads "$CP_ALL_IN_ONE_REPO" "$cp_post" >/dev/null 2>&1; then
+    echo "$cp_post"
+  elif git ls-remote --exit-code --heads "$CP_ALL_IN_ONE_REPO" "$cp_x" >/dev/null 2>&1; then
+    echo "$cp_x"
+  else
+    echo "master"
```
Comment (KrishVora01) [inline .semaphore/scripts/verify-cp-all-in-one.sh]: Qq: Should we fail if we dont find the corresponding .x branch or is defaulting to master the correct way here? Do you anticipate running tests against master to give a false signal in some edge case?

File: .semaphore/scripts/verify-cp-all-in-one.sh [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  echo "Using PR-built images:"
+  echo "  C3:   $dev_c3"
+  echo "  PROM: $dev_prom"
+  echo "  AM:   $dev_am"
+
+  rm -rf "$AIO_DIR"
+  git clone --depth 1 -b "$branch" "$CP_ALL_IN_ONE_REPO" "$AIO_DIR"
+  cd "$AIO_COMPOSE_DIR"
+
+  sed -i -E "s|image:[[:space:]]*confluentinc/cp-enterprise-control-center-next-gen:[^[:space:]]+|image: ${dev_c3}|g" docker-compose.yml
```
Comment (KrishVora01) [inline .semaphore/scripts/verify-cp-all-in-one.sh]: I was looking at what this command does using claude, it seems to say that:

> sed returns exit 0 when it matches nothing. So the moment cp-all-in-one changes anything about how that image line is written — a rename, a line wrap —
>   your three substitutions silently become zero. Then docker compose up happily pulls the public pinned :8.0.4 images, Control Center serves its 200, and the PR goes green without ever booting the PR's images.

File: .semaphore/scripts/verify-cp-all-in-one.sh (hunk 4 above)
Comment (rasikaAjoshi23) [reply .semaphore/scripts/verify-cp-all-in-one.sh]: +1 we should fail loud, but we should also make sure that it doesn't fail create and publish process of the image. 
Silently moving to master is not the right choice here.

Comment (riagarwal2002) [reply .semaphore/scripts/verify-cp-all-in-one.sh]: Sure, making the change

File: .semaphore/scripts/verify-cp-all-in-one.sh (hunk 5 above)
Comment (riagarwal2002) [reply .semaphore/scripts/verify-cp-all-in-one.sh]: Replaced the inline `sed` calls with a `patch_image` helper that asserts 
(a) the expected `image:` line exists before substitution and 
(b) the dev ECR URL is in the file after, failing with a clear error if either check misses.

**Review by rasikaAjoshi23 — APPROVED** [review]
LGTM
